### PR TITLE
BD-3044: Push subscription states

### PR DESCRIPTION
**Pull request by internetisaiah**
# Pull Request/Issue Resolution

#### Description of Change:
Updating the [Push subscription states](https://www.braze.com/docs/user_guide/message_building_by_channel/push/users_and_subscriptions/#push-sub-states) section. From Ticket:

```
Braze auto-opts-in when device push permission goes from disabled to enabled:
https://www.braze.com/docs/user_guide/message_building_by_channel/push/users_and_subscriptions/#update-push-subscription-state

We also auto-update push subscription status to "opted in" if a user re-enables push in device settings. this behavior can be disabled via:

https://www.braze.com/docs/user_guide/message_building_by_channel/push/users_and_subscriptions/#update-push-subscription-state

Configured via braze.xml through:

<bool name="com_braze_optin_when_push_authorized">true</bool>.

This adds the optInWhenPushAuthorized configuration property to specify whether a user's notification subscription state should automatically be set to optedIn when updating push permissions to authorized.

https://github.com/braze-inc/braze-swift-sdk/releases/tag/7.5.0
```

- Closes [BD-3044](https://jira.braze.com/browse/BD-3044)

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [x] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @internetisaiah as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Ana Teresa Serafino on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @internetisaiah for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@internetisaiah</td>
    <td>Developer Guide<br>Technology partners<br>Channels</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>Frontend Infrastructure & Experience (FIX)</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure) (tag @rachel-feinberg for Liquid use cases)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

**Comment by internetisaiah** [comment]
hey @davidbielik , was having a little trouble breaking down the request. let me know if/what changes should be made so i'm better aligned with your ask. thanks!

File: _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md [hunk 1]
```diff
@@ -38,11 +38,29 @@ Braze does not automatically change a user's push subscription state to `Unsubsc
...
-2. **REST API**<br>Use the [`/users/track` endpoint][users-track] to update the [`push_subscribe`][user_attributes_object] attribute for a given user.<br><br>
-3. **Automatically during opt-in** <br>When a user accepts the native OS push permission prompt, Braze will automatically change that user's subscription state to `Opted-In`.
+#### Automatic opt-in (default)
+
+By default, Braze sets a user's push subscription state to `Opted-In` when they first authorize push notifications for your app. Braze also does this when a user re-authorizes your permissions after previously revoking them.
+
```
Comment (davidbielik) [inline _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: this is an android config

File: _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md [hunk 2]
```diff
@@ -38,11 +38,29 @@ Braze does not automatically change a user's push subscription state to `Unsubsc
...
-2. **REST API**<br>Use the [`/users/track` endpoint][users-track] to update the [`push_subscribe`][user_attributes_object] attribute for a given user.<br><br>
-3. **Automatically during opt-in** <br>When a user accepts the native OS push permission prompt, Braze will automatically change that user's subscription state to `Opted-In`.
+#### Automatic opt-in (default)
+
+By default, Braze sets a user's push subscription state to `Opted-In` when they first authorize push notifications for your app. Braze also does this when a user re-authorizes your permissions after previously revoking them.
```
Comment (davidbielik) [inline _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:45]:
```suggestion
By default, Braze sets a user's push subscription state to `Opted-In` when they first authorize push notifications for your app. Braze also does this when a user re-enables push permissions in system settings after previously disabling them.
```

**Review by davidbielik — APPROVED** [review]

File: _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md (hunk 1 above)
Comment (internetisaiah) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: Sorry, I'm not tracking here. Does this mean this feature is only available on android? or does it just mean that it's available on ios, android, web but its not called `Opted-In` for the other platforms?

Comment (davidbielik) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: I just meant the line below  (47-50) (braze.xml) is android specific

Comment (internetisaiah) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: ahh, i see. is there a way to disable this in swift or web as well? i can add tabs if you give me the property syntax and file?

Comment (davidbielik) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: yea you already have that on line 54

Comment (internetisaiah) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: ah got it. what would the file for that xcode be? the `AppDelegate.swift` file? and what would be the syntax for the property?

Comment (davidbielik) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: i think it'd be used here: https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/a2-configure-braze

Comment (internetisaiah) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: how's this?
https://braze-docs-733qnhg05-braze.vercel.app/docs/user_guide/message_building_by_channel/push/users_and_subscriptions/#update-push-subscription-state

**Comment by internetisaiah** [comment]
@cla-bot check

File: _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md (hunk 1 above)
Comment (davidbielik) [reply _docs/_user_guide/message_building_by_channel/push/users_and_subscriptions.md:46]: thanks!

**Pull request merged**